### PR TITLE
AssemblyVersion to only use MajorVersion as per dotnet guidelines

### LINKED ISSUE #1695: Assembly versioning is not following recommended guidelines
https://docs.microsoft.com/dotnet/standard/library-guidance/versioning#assembly-version recommends assembly name to be Majorversion.0.0.0. The tool we use (https://www.nuget.org/packages/MinVer) already complies with this recommendation.

However we override the assembly version here : https://github.com/open-telemetry/opentelemetry-dotnet/blob/master/build/Common.prod.props#L25-L26

Proposal:
Remove the overriding, and simply use minver defaults, to get recommended versioning.

### PR DESCRIPTION
**Pull request by cijothomas**
Fixes #1695 

Also see: https://github.com/open-telemetry/opentelemetry-dotnet/issues/1696

## Changes

Please provide a brief description of the changes here.

For significant contributions please make sure you have completed the following items:

* [ ] `CHANGELOG.md` updated for non-trivial changes
* [ ] Design discussion issue #
* [ ] Changes in public API reviewed

### CODE REVIEW

File: build/Common.prod.props [hunk 1]
```diff
@@ -23,7 +23,6 @@
     <PropertyGroup>
       <RevisionNumber>0</RevisionNumber>
       <RevisionNumber Condition="$(MinVerVersion.Split(`.`).Length) == 4">$(MinVerVersion.Split(`.`)[3])</RevisionNumber>
-      <AssemblyVersion>$(MinVerMajor).$(MinVerMinor).$(MinVerPatch).$(RevisionNumber)</AssemblyVersion>
       <Version>$(MinVerMajor).$(MinVerMinor).$(MinVerPatch).$(RevisionNumber)</Version>
       <FileVersion>$(MinVerMajor).$(MinVerMinor).$(MinVerPatch).$(RevisionNumber)</FileVersion>
```
Comment (cijothomas) [inline build/Common.prod.props:27]: This target is still required, as the default Minver does not include revision number in the FileVersion. The dotnet recommendation is to *include* revision number - https://docs.microsoft.com/en-us/dotnet/standard/library-guidance/versioning#assembly-file-version

**Review by CodeBlanch — APPROVED** [review]
LGTM

**Review by eddynaka — APPROVED** [review]

**Review by reyang — APPROVED** [review]

**Comment by reyang** [comment]
Some existing code logic might be affected, and need to switch from Assembly Version to File Version.
For example: https://github.com/open-telemetry/opentelemetry-dotnet/blob/b710c314ae041ad5db655cce4547f921b88e7495/src/OpenTelemetry/Resources/ResourceBuilderExtensions.cs#L27

**Review by alanwest — APPROVED** [review]
LGTM. Per discussion with Cijo, I'm pushing the merge button to test things out.

**Pull request merged**